vunit/test/unit/test_ui.py

```diff
@@ -255,6 +255,52 @@ def test_add_source_files(self):
         for file_name in files:
             lib.get_source_file(file_name).name.endswith(file_name)
 
+    def test_add_source_files_from_csv(self):
+        csv = """
+        lib,  tb_example.vhdl  
+        lib1 , tb_example1.vhd
+        lib2, tb_example2.vhd
+        """
+
+        libraries = ['lib', 'lib1', 'lib2', 'lib2']
+        files = ['tb_example.vhdl', 'tb_example1.vhd', 'tb_example2.vhd', ' tb,ex3.vhd']
+        
+        self.create_csv_file('test_csv.csv', csv)
+        for file_name in files:
+            self.create_file(file_name) 
+        
+        ui = self._create_ui()
+        ui.add_source_files_from_csv('test_csv.csv')
+        
+
+        for i in range(len(libraries)):
+            library_name = libraries[i]
+            file_name = files[i]
+            file_name_from_ui = ui.add_source_files_from_csv(file_name, library_name)
+            self.assertIsNotNone(file_name_from_ui)
+        
+    def test_add_source_files_from_csv_return(self):
+        csv = """
+        lib, tb_example.vhd
+        lib, tb_example1.vhd
+        lib, tb_example2.vhd
+        """ 
+
+        list_of_files =  ['tb_example.vhd', 'tb_example1.vhd', 'tb_example2.vhd']
+        
+        for i in range(len(list_of_files)):
+            self.create_file(list_of_files[i], str(i)) 
+        
+        self.create_csv_file('test_returns.csv', csv)
+        ui = self._create_ui()
+
+        source_files = ui.add_source_files_from_csv('test_returns.csv')
+        
+        for i in range(len(source_files)):
```

Comment:
This test is flawed. source_files could just be empty or miss the last element and the test would still pass. Use list_of_files as the iteration length source instead. Or even better just compare the lists directly with` self.assertEqual([source_file.name for source_file in source_files, list_of_files)`